src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,276 @@
+    const bounceValue = useRef(new Animated.Value(1)).current
+
+    useEffect(() => {
+      setChecked(isChecked ?? false)
+    }, [isChecked])
+
+    const bounceInEffect = () => {
+      Animated.spring(bounceValue, {
+        toValue: bounceEffectIn,
+        velocity: bounceVelocityIn,
+        bounciness: bouncinessIn,
+        useNativeDriver,
+      }).start()
+    }
+
+    const bounceOutEffect = () => {
+      Animated.spring(bounceValue, {
+        toValue: bounceEffectOut,
+        velocity: bounceVelocityOut,
+        bounciness: bouncinessOut,
+        useNativeDriver,
+      }).start()
+    }
+
+    const syntheticBounceEffect = useCallback(() => {
+      Animated.sequence([
+        Animated.timing(bounceValue, {
+          toValue: bounceEffectIn,
+          duration: 50,
+          useNativeDriver,
+        }),
+        Animated.spring(bounceValue, {
+          toValue: bounceEffectOut,
+          velocity: bounceVelocityOut,
+          bounciness: bouncinessOut,
+          useNativeDriver,
+        }),
+      ]).start()
+    }, [bounceValue, bounceEffectIn, bounceEffectOut, bounceVelocityOut, bouncinessOut, useNativeDriver])
+
+    const renderCheckIcon = () => {
+      const checkStatus = disableBuiltInState ? isChecked : checked
+
+      return (
+        <IconContainer
+          testID={'icon-container'}
+          size={size}
+          backgroundColor={checked ? fillColor : unfillColor}
+          disable={disable}
+          disableOpacity={disableOpacity}
+          style={StyleSheet.flatten([{transform: [{scale: bounceValue}]}, iconStyle])}>
+          <InnerIconContainer style={innerIconStyle} size={size} borderColor={fillColor}>
+            {iconComponent ||
+              (checkStatus && <StyledImage source={checkIconImageSource} style={iconImageStyle} />)}
+          </InnerIconContainer>
+        </IconContainer>
+      )
+    }
+
+    const renderCheckboxText = () => {
+      const checkDisableTextType = typeof disableText === 'undefined'
+      return (
+        (!disableText || checkDisableTextType) &&
+        (textComponent || (
+          <TextContainer style={textContainerStyle} disable={disable} disableOpacity={disableOpacity}>
+            <Text testID="text" style={textStyle}>
+              {text}
+            </Text>
+          </TextContainer>
+        ))
+      )
+    }
+
+    const onHandlePress = useCallback(() => {
+      if (!disableBuiltInState) {
+        setChecked(prev => !prev)
+      }
+      syntheticBounceEffect()
+      onChange && onChange(!checked)
+    }, [disableBuiltInState, checked, onChange, syntheticBounceEffect])
+
+    useImperativeHandle(forwardedRef, () => ({onHandlePress}), [onHandlePress])
+
+    return (
+      <Container
+        testID="container"
+        {...rest}
+        style={style}
+        onPressIn={!disable ? bounceInEffect : null}
+        onPressOut={!disable ? bounceOutEffect : null}
+        onPress={!disable ? onHandlePress : null}>
```

Comment:
Why we don't pass direct `disabled` to component?

Reply:
updated